checker/src/main/java/org/checkerframework/checker/index/samelen/SameLenQualifier.java

```diff
@@ -0,0 +1,208 @@
+        return builder.build();
+    }
+
+    /** @return The list of sequences with an offset of zero. */
+    public List<String> sameLenArrays() {
+        return sameLenArrays("0");
+    }
+
+    /** @return The list of sequences with an offset of the given string. */
+    public List<String> sameLenArrays(String s) {
+        List<String> result =
+                sequences
+                        .stream()
+                        .filter(sequence -> offsets.get(sequence).toString().equals(s))
+                        .collect(Collectors.toList());
+        return result;
+    }
+
+    /**
+     * Returns true iff the given offset plus the value of the offset in this SameLen equation equal
+     * zero.
+     */
+    public boolean isSameLenByOffset(String sequence, OffsetEquation offset) {
+        return offset == null || !offsets.containsKey(sequence)
+                ? false
+                : offset.copyAdd('+', offsets.get(sequence)).equals(OffsetEquation.ZERO);
+    }
+
+    /** Returns true iff every sequence in the superType has the same offset in this qualifier. */
+    public boolean isSubtype(SameLenQualifier superType) {
+        for (String superArray : superType.sequences) {
+            if (!superType.offsets.get(superArray).equals(offsets.get(superArray))) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * If the set of sequences in other and in this qualifier are disjoint, returns unknown.
+     *
+     * <p>Otherwise, returns a new SameLenQualifier with the intersection of the two. Unknown is
+     * also returned if the same sequence has different offsets in the two qualifiers.
+     */
+    public AnnotationMirror lub(
+            SameLenQualifier other, AnnotationMirror unknown, ProcessingEnvironment env) {
+        if (!Collections.disjoint(this.sequences, other.sequences)) {
+            List<String> intersection = new ArrayList<>();
+            intersection.addAll(sequences);
+            intersection.retainAll(other.sequences);
+            List<String> intersectionOffsets = new ArrayList<>();
+            for (String sequence : intersection) {
+                if (offsets.get(sequence).equals(other.offsets.get(sequence))) {
+                    intersectionOffsets.add(offsets.get(sequence).toString());
+                } else {
+                    return unknown;
+                }
+            }
+            return of(intersection, intersectionOffsets).convertToAnnotation(env);
+        } else {
+            return unknown;
+        }
+    }
+
+    /**
+     * If the sequences are disjoint, returns bottom. Otherwise, returns the a combined SameLen
+     * qualifier.
+     */
+    public AnnotationMirror glb(
+            SameLenQualifier other, AnnotationMirror bottom, ProcessingEnvironment env) {
+        if (!Collections.disjoint(sequences, other.sequences)) {
+            return this.combine(other).convertToAnnotation(env);
```

Comment:
Will this trigger the assert in combine?  `@SameLen("a", offset = "b")` `@SameLen("a")`